renderer.php

```diff
@@ -133,12 +135,19 @@ function header($text, $level, $pos) {
             }
         }
 
-        if($level == 2){
+        if($level <= $this->getConf('maxHeaderLevelForNewSlide')){
             if($this->slideopen){
                 $this->doc .= '</div>'.DOKU_LF; //close previous slide
             }
             $this->doc .= '<div class="slide">'.DOKU_LF;
             $this->slideopen = true;
+
+            if ($level == 2) {
+                $this->lastH2 = $text;
+            } else {
+                $level = 2;
+                $text = $this->lastH2 . " - " . $text;
+            }
```

Comment:
Can you explain what this block is supposed to do?

Reply:
I was wondering the same thing. As I said, this is not my patch, this is something somebody dropped anonymously on dokuwiki.org a while ago.

I removed the definition of the lastH2 variable and removed the block on a branch of my fork. I applied it to my wikis running dokuwiki (All with very different settings) and the plugin works as it did before. I am guessing it served a purpose for the poster?

Anyways, sorry for being inattentive there. I am gonna open up a different pull request and close this one...